packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -337,6 +344,9 @@ export class BlobManager {
                     // no need to submit BlobAttach op in this case
                     entry.handleP.resolve(this.getBlobHandle(response.id));
                     this.pendingBlobs.delete(localId);
+                    if (!this.hasPendingBlobs()) {
```

Comment:
Might be nice to add a helper function that deletes an ID from pending and then emits if empty